### ISSUE TITLE
Incorrect behavior of "fittable" checkbox in polydispersity tab

### ISSUE THREAD

**Issue by butlerpd** [SasView/sasview#1522]
Once polydispersity is turned on, one can visit the polydispersity tab and choose the function and all the other parameters including the value of the PD parameter itself.  One can also choose to keep it fixed to that value during the fitting or let the fitting fit the polydispersity as well by checking or unchecking the checkbox.

However, checking and unchecking the box seems to be recognized as a change that need recomputing.  if one checks the box and then computes the curve is computed with something like a PD of more than 1. If it is unchecked and compute hit again it computes as if PD has been set to zero even if that is not true.

Simple example to demonstrate:
open a new fit panel and choose sphere and sphere.  Change the fit options to only go to 0.005 min q and log space 150 points. Now back to Model tab and compute then show plot.  We have the typical sphere form factor with no polydispersity smearing. Now check polydispersity and go to that tab
make PD=0.3 and compute. the oscillations get significantly damped as they should. Now check the fittable box (which means nothing for a theory curve but I can so I try it of course :-) and the Show plot turns to compute. So hit compute.  the cuve now goes almost to a straight line which from what I can tell is applying a polydispersity of between 1 and 2 .... .even though the PD is STILL set at 0.3.  type 0.3 and return and nothing happens because no change is detected. Now uncheck the box and again compute option appears.  Hit it and the original form factor with all the oscillations appears (i.e. as if PD=0 ... even though it is still 0.3).  The only way to get 0.3 back is to type in 0.29 o 0.31 or something close but different.  Then compute appears and the correct curve will be computed.

For fun you can now add a data set (say polyspheres.txt) the behavior is essentially the same. But worse now, checking the box again and fitting does seem to compute the proper polydispersity (using DREAM you can see where the center is that it calculates) but that does not get passed back somehow and the curve is using either the very large distribution (when checked) or zero when unchecked regardless of the PD value.... until a small change is made to that value to reset it. Using LM, the Chi^2 seems to be correct though using DREAM, while the DREAM distribution plots show the correct PD, the Chi^2 seem to be huge ... at least sometimes?

Bottom line is that I would say that polydispersity seems to be completely unusable at the moment?

**Comment by butlerpd** [comment]
This one is more critical but I still don't believe it is new (by the nature of what it is doing it should not be new).  I thus think we should probably move this to 5.0.3?  I guess we should double check that it behaves badly in 5.0.1 as well....?

**Comment by m2cci-NMZ** [comment]
The problem seems to come from the `onPolyModelChange` method from the `FittingWidget` class
The value of the polydispersity was set to:
```
self.poly_params[key] = value
```
with `value` variable defined as the checkstate of the checkbox
```
value = item.checkState()
```
So the value of the polydispersity was being changed to the value of the checkbox state (0 when unchecked, 2 when checked) every time the checkbox was checked/unchecked
